omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/VisitController1_9Test.java

```diff
@@ -214,4 +214,14 @@ public void searchByPatient_shouldGetUnretiredVisitsForThePatient() throws Excep
 		Assert.assertEquals(3, Util.getResultsSize(result));
 	}
 	
+	@Test
+	public void searchByPatient_shouldGetAllVisitsForThePatientIncludingRetired() throws Exception {
+		handle(newDeleteRequest(getURI() + "/" + getUuid(), new Parameter("reason", "void test reason")));
+		SimpleObject resultWithVoidedVisits = deserialize(handle(newGetRequest(getURI(), new Parameter("patient",
+		        "da7f524f-27ce-4bb2-86d6-6d1d05312bd5"), new Parameter("includeAll", "true"))));
+		SimpleObject resultVithoutVoidedVisits = deserialize(handle(newGetRequest(getURI(), new Parameter("patient",
+		        "da7f524f-27ce-4bb2-86d6-6d1d05312bd5"))));
+		Assert.assertNotEquals(Util.getResultsSize(resultWithVoidedVisits), Util.getResultsSize(resultVithoutVoidedVisits));
```

Comment:
Since you are voiding only one visit, shouldn't this assertion pass?
Assert.assertEquals(Util.getResultsSize(resultWithVoidedVisits) - 1, Util.getResultsSize(resultVithoutVoidedVisits));

Reply:
The thing is, based on the dataset provided, one visit was voided (line 188)before making the void call, making it two (2) voided visits, so it will not pass. Peeking into the results, resultsWithvoidedVisits has 4 visits while resultsWithvoidedVisits has 2.

Reply:
It looks strange for a test to void one visit and expect two to be voided. Can you fix it by asserting that we had one voided visit before this one?

Reply:
I have proved that beside the one I voided (getUuid == 1e5d5d48-6b78-11e0-93c3-18a905e044dc),e1428fea-6b78-11e0-93c3-18a905e044dc is also voided before my test.

Reply:
Your change does not account for the 4 vs 2 records after voiding only one.